### PR TITLE
Keypair

### PR DESCRIPTION
**Pull request by EarthmanT**

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -1,4 +1,3 @@
--e git+https://github.com/cloudify-cosmo/cloudify-dsl-parser@master#egg=cloudify-dsl-parser==3.2a2
```
Comment (nir0s) [inline dev-requirements.txt]: dude.. you gotta stop changing the same things in different branches. You're gonna lose your mind and make mistakes merging these.

File: ec2/keypair.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    key_pair_name = ctx.node.properties['name']
+    ctx.logger.info('Creating key pair.')
+
+    try:
+        kp = ec2_client.create_key_pair(key_pair_name)
+    except (EC2ResponseError, BotoServerError, BotoClientError) as e:
+        raise NonRecoverableError('Key pair not created. '
+                                  'API returned: {0}'.format(e))
+
+    ctx.logger.info('Created key pair.')
```
Comment (nir0s) [inline ec2/keypair.py]: key pair: {0}

File: ec2/utils.py [hunk 3]
```diff
@@ -120,3 +120,21 @@ def get_instance_from_id(instance_id, ctx):
...
+
+    try:
+        key_pair_object.save(ctx.node.properties['private_key_path'])
+    except OSError:
+        raise NonRecoverableError('Unable to save key pair to file: {0}.'
+                                  'OS Returned: {1}'.format(
+                                      ctx.node.properties['private_key_path'],
+                                      OSError))
+
+    ctx.logger.debug('No errors were raised while saving the '
```
Comment (nir0s) [inline ec2/utils.py]: ?

File: ec2/keypair.py (hunk 2 above)
Comment (EarthmanT) [inline ec2/keypair.py]: Added to each method. where an operation completes successfully.

File: ec2/utils.py (hunk 3 above)
Comment (EarthmanT) [inline ec2/utils.py]: Ok. I removed it.

**Pull request merged**